### ISSUE TITLE
Code for sorting seems to fail in Laravel 5.4

### ISSUE THREAD

**Issue by Quibi** [dimsav/laravel-translatable#362]
Hi,

I am running a Laravel 5.4 site and I have tried to sort by translations as documentation states. It has failed in my case and has returned an empty collection. No translations found although they really exist (I can retrieve them with all() and get() model methods).

I tried to run something similar to:

```
Country::join('country_translations as t', 't.country_id', '=', 'countries.id')
    ->where('locale', 'en')
    ->groupBy('countries.id')
    ->orderBy('t.name', 'desc')
    ->with('translations')
    ->get();
```

as you suggest in readme.

After some tests, my "solution" consists in using whereRaw instead of where.

```
Country::join('country_translations as t', 't.country_id', '=', 'countries.id')
    ->whereRaw('locale = "en"')
    ->groupBy('countries.id')
    ->orderBy('t.name', 'desc')
    ->with('translations')
    ->get();
```

Also I have to change `'strict' => true` to `'strict' => false` in `config/database.php` in order to avoid that Laravel raises an error.

With these changes, the expected results were returned.

Maybe I am doing something wrong —I am a newbie—, maybe something has changed in Laravel 5.4 (so many things have actually changed without backwards compatibility).

Just for your consideration in case it will be useful. Thanks.

**Comment by daviskoko** [comment]
Actually I don't use this package, but your query doesn't seem to me like it will return any thing. You're trying to mix query builder with eloquent. 
Rewrite your query as such using query builder:
```
$countries = DB::table('countries')
                            - >join('country_translations', 'countries.id', '=', 'country_translations.id') 
                          - >where('country_translations.locale', 'en') 
                          - >orderBy('country translations.name', 'desc')
                          - >get();
```

**Comment by unitedworx** [comment]
Here is an example code I have working with sorting

    $features = Feature::join('feature_translations as translations', function ($join) {
						            $join->on('features.id', '=', 'translations.feature_id')
						                 ->where('translations.locale', '=', 'en');
						        })        
						        ->orderBy('translations.name', 'asc')
						        ->select('features.*') 
						        ->with('translations')
						->get();

**Comment by unitedworx** [comment]
P.S. your initial code does not work since your where clause is bound to your country table intead of your join. Your where raw probably works since it’s right after the join, I wouldn’t keep it, I am surprised it even works :)

**Comment by Quibi** [comment]
@daviskoko I should point it out that it is not my code. It comes from the documentation of this package. I really do not know if has worked for anyone else or if there is a bug in the documentation or, most probably (in my opinion) it fails under Laravel 5.4.

However, I must say that I have tried something similar to your proposal (use a query builder instead of the model) and it does not work at all. Maybe your code is right and I have made another mistake. Sorry about that and thanks.

@unitedworx I will test your code tomorrow (it is late here). 

I think that your assumption that the where method is failing in "my" code because is bound to the country table is right. (Can be this the reason why @daviskoko probably does not work either?)

Indeed, if you ask for the toSql() of 'my' query you can see that the sql where clause is something like `where locale = ?` instead of `where locale = 'en'`. I do not know what that means but I suspect that it has something related to the fact that where method is bound to the country table instead of the join.

Of course, using a raw where is neither a 'nice' nor a 'trustworthy' solution. But, as a patch-up, it does its job because the underlying sql is correct.

As I said, I will test your proposal tomorrow and report the results.

Thank you so.

**Comment by daviskoko** [comment]
@Quibi Regardless of the laravel version, query builder will work. Using eloquent is largely dependent on the relationships that the programmer set in the model. The reason I used query builder is because am not sure of what the relationship in the model looks like. Query builder will bypass the model relationship & focus on table relationship, also its closer to SQL commands than Eloquent. So if my code does not work, then there is something wrong in the table schema you're using.

**Comment by unitedworx** [comment]
yeah the code sample in the documentation is buggy :) not related to laravel version number

**Comment by Quibi** [comment]
@daviskoko I am really a newbie in Laravel and my knowledge about this framework is very little at this point.

I tried to follow your sample in the documentation as close as I can. But, of course, as you pointed out, my schema is probably wrong despite my efforts. I have tried really hard to find where I have made a mistake but I am just a newbie, you know. 

Indeed, the only thing I can say for sure is that, in my case —just in my case—, the code proposed by @unitedworx works and I think that maybe it worth a deep look. That is all.

Thanks for your attention.
